src/main/java/zowe/client/sdk/zosjobs/SubmitJobs.java

```diff
@@ -235,39 +232,30 @@ public Job submitJobCommon(SubmitJobParams params) throws Exception {
     /**
      * Parse input string for JCL substitution
      *
-     * @param symbols JCL substitution symbols e.g.: "SYMBOL=SYM SYMBOL2=SYM2"
+     * @param keyValues Map containing JCL substitution symbols e.g.: "{"SYMBOL","SYM"},{"SYMBOL2","SYM2"}
      * @return String Map containing all keys and values
      * @throws Exception error on submitting
      * @author Corinne DeStefano
      */
-    private Map<String, String> getSubstitutionHeaders(String symbols) throws Exception {
-        var symbolMap = new HashMap<String, String>();
-
-        // Input range: KEY="abc cde fg" KEY2="none" KEY3=none KEY4="nospace" etc
-        // REGEX: (\w*)=("(.*?)"|\S*)
-
-        // Check overall structure of input string
-        List<String> pairs = List.of(symbols.split(" "));
-        for (String pair : pairs) {
-            if (pair.split("=").length % 2 != 0) {
-                throw new Exception("Encountered invalid key/value pair. " +
-                        "Use the format KEY=VALUE separated by space for each pair in string");
-            }
-        }
+    public Map<String, String> getSubstitutionHeaders(Map<String, String> keyValues) throws Exception {
+
+        Map<String,String> symbolMap = new HashMap<>();
 
         // Check for matching quotes
-        if (symbols.chars().filter(ch -> ch == '"').count() % 2 != 0) {
-            throw new Exception("Encountered invalid key/value pair. Mismatched quotes.");
+        for (String value : keyValues.values()) {
+            if (value.chars().filter(ch -> ch == '"').count() % 2 != 0) {
+                throw new Exception("Encountered invalid key/value pair. Mismatched quotes.");
+            }
+
+            if (value.length() == 0) {
+                throw new Exception("Encountered invalid key/value pair. Must define a value for key/value pairs.");
+            }
         }
 
-        // Now get the groups
-        Pattern pattern = Pattern.compile("(\\w*)=(\"(.*?)\"|\\S*)");
-        Matcher matcher = pattern.matcher(symbols);
-        while (matcher.find()) {
-            String key = matcher.group(1);
-            String value = matcher.group(2);
+        for (String key: keyValues.keySet()) {
+            String value = keyValues.get(key);
 
-            if (key.length() == 0 || value.length() == 0) {
+            if (key.length() == 0) {
                 throw new Exception("Encountered invalid key/value pair. Must define a value for key/value pairs.");
```

Comment:
This message should say "Must define a key for key/value pair." ? use key and singular pair